src/subtitles-octopus.js

```diff
@@ -18,25 +34,36 @@ var SubtitlesOctopus = function (options) {
 
     self.timeOffset = options.timeOffset || 0; // Time offset would be applied to currentTime from video (option)
 
-    if (typeof ImageData.prototype.constructor !== 'function') {
-        (function () {
-            var canvas = document.createElement('canvas');
-            var ctx = canvas.getContext('2d');
+    self.hasAlphaBug = false;
+
+    (function() {
```

Comment:
Why is this in a self-executing function?

Reply:
Maybe extract it too

Reply:
Again, this is within a nested function to not pollute the global namespace. It has single use, thus it's nameless and executed right away.